docs/source/guides/networking.rst

```diff
@@ -0,0 +1,144 @@
+	}
+
+	#if CLIENT
+	void function Server_GetNetworkedVariable( int number ) // you can declare as many or few parameters as you wish
+	{
+		printt("got integer", number)
+	}
+	#endif
+
+Calling the ``CLIENT`` function ``Server_GetNetworkedVariable`` on ``SERVER`` vm:
+
+.. code-block:: javascript
+
+	// player: CPlayer entity that should execute the function
+	// func: function identifier string
+	// ...: any parameters passed to the function
+	Remote_CallFunction_NonReplay( entity player, string func, ... ) // NOT reexecuted in a replay
+	Remote_CallFunction_Replay( entity player, string func, ... ) // reexecuted in a replay
+
+	// for the previous example, this would be a valid remote function call:
+
+	Remote_CallFunction_NonReplay( player, "Server_GetNetworkedVariable", RandomIntRange( 1, 100 ) )
+
+Server to Client command callbacks
+----------------------------------
+
+Allows the ``SERVER`` vm to create a ``ServerToClientStringCommand`` on a player which is linked to a Callback locally
+
+Register a server command
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. note:: 
+
+	this has to be executed on the ``Before`` Client callback
+
+	the formatting for the server command is like a normal console command. Arguments are seperated by spaces
+
+Register with the ``AddServerToClientStringCommandCallback( string func, void functionref( array<string> ) reference )`` function clientside and execute with the ``ServerToClientStringCommand( entity player /*CPlayer*/, string command )`` function server side
+
+Example:
+^^^^^^^^
+
+.. code-block:: javascript
+
+	void function MessageUtils_ClientInit()
+	{
+		AddServerToClientStringCommandCallback( "ServerHUDMessageShow", ServerCallback_CreateServerHUDMessage )
+	}
+
+	void function ServerCallback_CreateServerHUDMessage ( array<string> args )
+	{
+		// client side command handle logic ...
+	}
+
+
+``CLIENT`` to ``SERVER`` vm
+===========================
+
+Client to Server command callbacks
+----------------------------------
+
+Register a client command callback serverside with ``AddClientCommandCallback( string command, bool functionref( entity player /*CPlayer*/, array<string> args ) callback )``
+
+``player`` is the player that called the command client side. The callback function should return ``true`` if the command was accepted and ``false`` if it was invalid.
+
+The ``CLIENT`` vm can execute commands with the ``player.ClientCommand( string command )`` function. These will be handled by the ``SERVER`` if the command is registered.
+
+
+``CLIENT`` to ``UI`` vm
+=======================
+
+Create a global function in the ``UI`` vm and call it in the ``CLIENT`` vm with the ``RunUIScript( string identifier, ... )`` function. You can also pass parameters to the function.
+
+``UI`` to ``CLIENT`` vm
+=======================
+
+Create a global function in the ``CLIENT`` vm and call it in the ``UI`` vm with the ``RunClientScript( string identifier, ... )`` function. You can also pass parameters to the function.
```

Comment:
Might also be good to provide some examples here.
What is `identifier`? I know this, but users reading this might not